scrapy_zyte_api/responses.py

```diff
@@ -0,0 +1,123 @@
+from base64 import b64decode
+from typing import Dict, List, Optional, Union, Tuple
+
+from scrapy import Request
+from scrapy.http import Response, TextResponse
+from scrapy.responsetypes import responsetypes
+
+_DEFAULT_ENCODING = "utf-8"
+
+
+class ZyteAPIMixin:
+
+    REMOVE_HEADERS = {
+        # Zyte API already decompresses the HTTP Response Body. Scrapy's
+        # HttpCompressionMiddleware will error out when it attempts to
+        # decompress an already decompressed body based on this header.
+        "content-encoding"
+    }
+
+    def __init__(self, *args, zyte_api: Dict = None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._zyte_api = zyte_api
+
+    @property
+    def zyte_api(self) -> Optional[Dict]:
```

Comment:
Naming :) What do you think about calling it with something "raw" in the name? raw_api_response, raw?

I'm still unsure about `response.zyte_api`: response is already a Zyte API response, it's just presented in a different way, mimicking Scrapy response. The difference is not that .zyte_api is a Zyte API resonse, and response is not, the difference is in the format of data; .zyte_api contains the raw, unprocessed version, with all the information available.

Reply:
`response.raw_api_response` then? Any shorter alternative that sounds nice? `response.raw_data`?

Reply:
A good point. I think `response.raw_api` might be good. I believe having the word _"api"_ presents a better context about what the user is trying to access. What do you think?

Reply:
I think that's fine for the attribute name to be a bit longer. It seems a concern about `response.raw_api_response` is that "response" is written two times, while "response.raw_api" could already read as "raw api response". But this logic doesn't work for other Response attributes, other `response` attributes are not responses themselves. Because of that, `response.raw_api_response` looks better to me than `response.raw_api`.

Reply:
`response.raw_data` also looks fine.

Reply:
> But this logic doesn't work for other Response attributes, other response attributes are not responses themselves. 

You've raised a good point. Renamed it to `response.raw_api_response`. https://github.com/scrapy-plugins/scrapy-zyte-api/pull/10/commits/37a4cc705d786a8d091b3066e6426b1f2dc9814c